tests/test_profile.py

```diff
@@ -0,0 +1,129 @@
+        self.p = Profile([[0.0, 5.0], [0.1, 10.0], [0.2, 20.0], [0.3, 10.0]])
+
+    def test_x_at_y(self):
+        # points outside profile range [5.0, 20.0]
+        self.assertTrue(np.isnan(self.p.x_at_y(-1.0)))
+        self.assertTrue(np.isnan(self.p.x_at_y(4.9999999)))
+        self.assertTrue(np.isnan(self.p.x_at_y(20.0000001)))
+        self.assertTrue(np.isnan(self.p.x_at_y(7.5, reverse=True)))
+        # known points
+        self.assertAlmostEqual(self.p.x_at_y(5.0), 0.0)
+        self.assertAlmostEquals(self.p.x_at_y(10.0), 0.1)
+        self.assertAlmostEquals(self.p.x_at_y(20.0), 0.2)
+        self.assertAlmostEquals(self.p.x_at_y(20.0, reverse=True), 0.2)
+        self.assertAlmostEquals(self.p.x_at_y(10.0, reverse=True), 0.3)
+        # some points between known ones
+        self.assertAlmostEquals(self.p.x_at_y(7.5), 0.05)
+        self.assertAlmostEquals(self.p.x_at_y(11.11), 0.1111)
+        self.assertAlmostEquals(self.p.x_at_y(19.99), 0.1999)
+        self.assertAlmostEquals(self.p.x_at_y(19.99, reverse=True), 0.2001)
+        # basic exception testing
+        with self.assertRaises(TypeError):
+            self.p.x_at_y()
+        with self.assertRaises(TypeError):
+            self.p.x_at_y('a')
+
+    def test_width(self):
+        # out of range
+        self.assertTrue(np.isnan(self.p.width(0.0)))
+        self.assertTrue(np.isnan(self.p.width(5.0)))
+        self.assertTrue(np.isnan(self.p.width(9.9)))
+        self.assertTrue(np.isnan(self.p.width(20.1)))
+        self.assertTrue(np.isnan(self.p.width(25.0)))
+        # in range
+        self.assertAlmostEquals(self.p.width(10.0), 0.2)
+        self.assertAlmostEquals(self.p.width(15.0), 0.1)
+        self.assertAlmostEquals(self.p.width(20.0), 0.0)  # only one point (0, 20)
+        # some exception testing
+        with self.assertRaises(TypeError):
+            self.p.width()
+        with self.assertRaises(TypeError):
+            self.p.width('a')
+
+    def test_fwhm(self):
+        p1 = Profile([[1, 1], [2, 2], [3, 1]])
+        self.assertAlmostEquals(p1.fwhm, 2)
+        p2 = Profile([[-12, 1], [-1, 17], [0, 3], [3, 1]])
+        self.assertAlmostEquals(p2.fwhm, 6.45089, 5)
+        # should go out of range
+        p3 = Profile([[-12, 1], [-1, 7], [0, 3], [3, 17]])
+        self.assertTrue(np.isnan(p3.fwhm))
+        # it is not callable
+        with self.assertRaises(TypeError):
+            self.p.fwhm()
+
+    def test_normalize(self):
+        p1 = Profile([[1, 1], [2, 20], [3, 40]])
+        p1.normalize(1)
+        self.assertTrue(np.array_equal(p1.y, [1, 20, 40]))
+
+        p1 = Profile([[1, 1], [2, 20], [3, 40]])
+        p1.normalize(2)
+        self.assertTrue(np.array_equal(p1.y, [0, 1, 3]))
+
+        p1 = Profile([[1, 1], [2, 20], [3, 40]])
+        p1.normalize(100)
+        self.assertTrue(np.array_equal(p1.y, [0, 0, 1]))
+
+        # case - less or equal to 0
+        p1 = Profile([[1, 1], [2, 20], [3, 40]])
+        p1.normalize(-1)
```

Comment:
Good idea. Such exception is OK, but add also some message in it, guiding user what type of input is expected.